src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp

```diff
@@ -0,0 +1,546 @@
+  __ movapd(xmm0, xmm17);
+  __ vmovsd(xmm21, xmm18, xmm20);
+  __ evfnmadd213sd(xmm0, xmm21, xmm22, Assembler::EVEX_RZ);
+//     while (bs2 <= a)
+  __ comisd(xmm0, xmm3);
+  __ jcc(Assembler::aboveEqual, L_11f4);
+  __ jmp(L_1231);
+//     }
+//   }
+//   else
+//   a = DP_FNMA_RZ(bs, q, a);
+  __ bind(L_1227);
+  __ movapd(xmm0, xmm4);
+  __ evfnmadd213sd(xmm0, xmm3, xmm17, Assembler::EVEX_RZ);
+
+//   while (bs <= a)
+  __ bind(L_1231);
+  __ comisd(xmm0, xmm2);
+  __ jcc(Assembler::below, L_128a);
+//   {
+//     q = DP_DIV_RZ(a, bs);
+  __ bind(L_1237);
+  __ vmovsd(xmm20, xmm18, xmm0);
+  __ evdivsd(xmm0, xmm20, xmm4, Assembler::EVEX_RZ);
+//     q = DP_ROUND_RZ(q);
+  __ vmovsd(xmm3, xmm18, xmm0);
+  __ vrndscalesd(xmm17, xmm18, xmm3, 0x0B);
+//     a = DP_FNMA_RZ(bs, q, a);
+  __ movapd(xmm0, xmm4);
+  __ vmovsd(xmm19, xmm18, xmm17);
+  __ evfnmadd213sd(xmm0, xmm19, xmm20, Assembler::EVEX_RZ);
+
+//   while (bs <= a)
+  __ comisd(xmm0, xmm2);
+  __ jcc(Assembler::aboveEqual, L_1237);
+  __ jmp(L_128a);
+//   // y is NaN?
+//   if (!(b <= DP_CONST(7ff0000000000000))) return y + y;
+__ bind(L_1268);
+__ vaddsd(xmm0, xmm3, xmm3);
+__ jmp(L_exit);
+//     return DP_FNMA(b, q, a);    // NaN
+
+  __ bind(L_1271);
+  __ vfnmadd213sd(xmm5, xmm4, xmm17);
+  __ movapd(xmm0, xmm5);
+  __ jmp(L_exit);
+
+//   a = DP_FNMA_RZ(b, q, a);
+  __ bind(L_1280);
+  __ movapd(xmm0, xmm5);
+  __ evfnmadd213sd(xmm0, xmm4, xmm17, Assembler::EVEX_RZ);
+
+// FMOD_CONT:
+//   while (b <= a)
+  __ bind(L_128a);
+  __ comisd(xmm0, xmm1);
+  __ jcc(Assembler::below, L_12bf);
+//   {
+//     q = DP_DIV_RZ(a, b);
+  __ bind(L_1290);
+  __ vmovsd(xmm17, xmm18, xmm0);
+  __ evdivsd(xmm0, xmm17, xmm5, Assembler::EVEX_RZ);
+//     q = DP_ROUND_RZ(q);
+  __ vmovsd(xmm2, xmm18, xmm0);
+  __ vrndscalesd(xmm3, xmm18, xmm2, 0x0B);
+//     a = DP_FNMA_RZ(b, q, a);
+  __ movapd(xmm0, xmm5);
+  __ vmovsd(xmm4, xmm18, xmm3);
+  __ evfnmadd213sd(xmm0, xmm4, xmm17, Assembler::EVEX_RZ);
+
+// FMOD_CONT:
+//   while (b <= a)
+//   }
+  __ comisd(xmm0, xmm1);
+  __ jcc(Assembler::aboveEqual, L_1290);
+
+//   a = DP_XOR(a, sgn_a);
+  __ bind(L_12bf);
+  __ vxorpd(xmm0, xmm0, xmm16, Assembler::AVX_512bit);
+
+//   return a;
+__ jmp(L_exit);
+
+//   if (!eq)  return x + sgn_a;
+  __ bind(L_12ca);
+  __ vaddsd(xmm0, xmm20, xmm16);
+  __ bind(L_exit);
+  __ addptr(rsp, 0x28);
+
+  } else if (VM_Version::supports_fma()) {       // AVX2 version
+
+    Label L_104a, L_11bd, L_10c1, L_1090, L_11b9, L_10e7, L_11af, L_111c, L_10f3, L_116e, L_112a;
+    Label L_1173, L_1157, L_117f, L_11a0;
+
+//   double fmod(double x, double y)
+// {
+// double a, b, sgn_a, q, bs, bs2, corr, res;
+// unsigned eq;
+// unsigned mxcsr, mxcsr_rz;
+
+//   __asm { stmxcsr DWORD PTR[mxcsr] }
+//   mxcsr_rz = 0x7f80 | mxcsr;
+  __ push(rax);
+  __ stmxcsr(Address(rsp, 0));
+  __ movl(rax, Address(rsp, 0));
+  __ movl(rcx, rax);
+  __ orl(rcx, 0x7f80);
+  __ movl(Address(rsp, 0x04), rcx);
```

Comment:
It may rarely happen that scope of MXCSR change is beyond couple of instruction, hence we simply load the needed settings and later on re-load std MXCSR settings from default location `ldmxcsr(ExternalAddress(StubRoutines::x86::addr_mxcsr_std()));`